bonsai/tests/behavior_tests.rs

```diff
@@ -410,3 +425,77 @@ fn test_after_all_succeed_out_of_order() {
     assert_eq!(s, Failure);
     assert_eq!(dt, 0.0);
 }
+
+
+#[test]
+fn test_repeat_sequence() {
+    {
+        let a: i32 = 0;
+        let after = RepeatSequence(Box::new(Action(LessThanRunningSuccess(5))),
+                                   vec![Action(Inc)]);
+
+        let mut state = State::new(after);
+
+        let (a, s, dt) = tick(a, 0.0, &mut state);
+
+        assert_eq!(a, 5);
+        assert_eq!(s, Success);
+        assert_eq!(dt, 0.0);
+
+        let (a, s, dt) = tick(a, 0.0, &mut state);
+
+        assert_eq!(a, 5);
+        assert_eq!(s, Success);
+        assert_eq!(dt, 0.0);
+    }
+}
+
+#[test]
+fn test_repeat_sequence_fail() {
+    {
+        let a: i32 = 4;
+        let after = RepeatSequence(Box::new(Action(LessThanRunningSuccess(5))),
+                                   vec![Action(Dec), Action(LessThan(0))]);
+        let mut state = State::new(after);
+        let (a, s, dt) = tick(a, 0.0, &mut state);
+
+        assert_eq!(a, 3);
+        assert_eq!(s, Failure);
+        assert_eq!(dt, 0.0);
+    }
+}
+
+#[test]
+fn test_repeat_sequence_timed() {
```

Comment:
I think this makes sense.

wrt to `dt`, when one action terminates, it can consume some of `dt` and the remaining is passed onto the next action.

Reply:
ref: https://github.com/Sollimann/bonsai/blob/main/docs/concepts/README.md#events

<img width="1416" alt="image" src="https://github.com/Sollimann/bonsai/assets/147421892/436051d1-f150-4612-a1c4-aa13af5c7c5e">